routes/profile.js

```diff
@@ -32,52 +32,64 @@ router.get('/:id/:profile', async (req, res) => {
  * Queries postgres for current, previous, and compare data for a given
  * profile type, set of geoids and compare geoid. Joins the data, and adds
  * 'change' and 'difference' calculation values.
- * @param {string} profile - The profile type
+ * @param {string} profileName - The profile type
  * @param {Array} geoids - The list of geoids for the given selected geography
  * @param {string} compare - Integer string representing the geoid of the comparison geography
  * @returns {Object}
  */
-async function getProfileData(profile, geoids, compare, db) {
+async function getProfileData(profileName, geoids, compare, db) {
   const isAggregate = geoids.length > 1;
 
-  const queryBuilder = getQueryBuilder(profile);
+  const queryBuilder = getQueryBuilder(profileName);
 
   // get data from postgres
   const [profileData, previousProfileData, compareProfileData] = await Promise.all([
-    db.query(queryBuilder(profile, geoids)),
-    db.query(queryBuilder(profile, geoids, /* is previous */ true)),
-    db.query(queryBuilder(profile, compare)),
+    db.query(queryBuilder(profileName, geoids)),
+    db.query(queryBuilder(profileName, geoids, /* is previous */ true)),
+    db.query(queryBuilder(profileName, compare)),
   ]);
 
   // create Dataframe from profile data
-  let profileDF = new DataIngestor(profileData, profile, isAggregate).processRaw();
+  let profileDF = new DataIngestor(
+    profileData,
+    profileName,
+    isAggregate,
+  ).processRaw();
 
   // join with previous profile data Dataframe
   profileDF = profileDF.join(
-    new DataIngestor(previousProfileData, profile, isAggregate, /* is previous */ true).processRaw('previous'),
+    new DataIngestor(
+      previousProfileData,
+      profileName,
+      isAggregate,
+      /* is previous */ true,
+    ).processRaw(),
     'variable',
   );
 
   // join with compare profile data Dataframe
   profileDF = profileDF.join(
     // 'compare' profiles are always NOT aggregate; they are comprised of a single geoid
-    new DataIngestor(compareProfileData, profile).processRaw('comparison'),
+    new DataIngestor(
+      compareProfileData,
+      profileName,
+      /* is aggregate */ false,
+      /* is previous */ false,
+      /* is comparison */ true,
+    ).processRaw(),
```

Comment:
One API enhancement we could do is change the parameters of `DataIngestor` to expect an object for _optional_ params like is aggregate, previous, comparision, etc:

```javascript
new DataIngestor(compareProfileData, profileName, {
  isAggregate: false,
  isPrevious: false,
  isComparison: true,
}).processRaw(),
```